core/examples/sync_ops.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+/*
+ This example shows you how to define ops in Rust
+ and then call them from JavaScript.
+*/
+
+use std::io::Write;
+
+use deno_core::{JsRuntime, Op, RuntimeOptions};
+use serde_json::json;
+
+pub fn main() -> Result<(), deno_core::error::AnyError> {
+  // Initialize a runtime instance
+  let mut runtime = JsRuntime::new(RuntimeOptions {
+    js_error_create_fn: None,
+    module_loader: None,
+    startup_snapshot: None,
+    will_snapshot: false,
+    heap_limits: None,
+  });
+
+  /*
+   The first thing we do is define two ops.
+   They will be used to show how to pass data
+   to Rust and back to JavaScript.
+
+   The first one is used to print data to stdout,
+   because by default the JavaScript console functions
+   are just stubs (they don't do anything).
+
+   The second one just transforms some input and returns it to JavaScript.
+  */
+
+  // Register the op for outputting bytes to stdout.
+  // It can be invoked with Deno.core.dispatch and the id this method returns
+  // or Deno.core.dispatchByName and the name provided.
+  runtime.register_op(
+    "op_print",
+    // The op_fn callback takes a state object OpState
+    // and a vector of ZeroCopyBuf's, which are mutable references
+    // to ArrayBuffer's in JavaScript.
+    |_state, zero_copy| {
+      let mut out = std::io::stdout();
+
+      // Write the contents of every buffer to stdout
+      for buf in zero_copy {
+        out.write_all(&buf).unwrap();
+      }
+
+      Op::Sync(Box::new([])) // No meaningful result
+    },
+  );
+
+  // Register the JSON op for summing a number array.
+  // A JSON op is just an op where the first ZeroCopyBuf is
+  // a serialized JSON value,
+  // the return value is also a serialized JSON value.
+  // It can be invoked with Deno.core.jsonOpSync and the name.
+  runtime.register_op("op_sum", |_state, zero_copy| {
+    // This helper function creates a JSON error result
+    // which throws an exception on the JavaScript side.
+    fn new_type_error(message: &str) -> Op {
+      // Create the value
+      let type_error_json = json!({
+        "err": {
+          "className": "TypeError",
+          "message": message,
+        }
+      });
+      Op::Sync(
+        // And serialize it
+        serde_json::to_vec(&type_error_json)
+          .unwrap()
+          .into_boxed_slice(),
+      )
+    }
+
+    // First we check that we got the JSON argument
+    if zero_copy.len() != 1 {
+      return new_type_error("Expected exactly one argument");
+    }
+
+    // Then we deserialize it
+    let json: serde_json::Value = match serde_json::from_slice(&zero_copy[0]) {
```

Comment:
This can be done automatically by using https://docs.rs/deno_core/0.58.0/deno_core/fn.json_op_sync.html